### PR TITLE
main: added structured log option [Fixes #39]

### LINKED ISSUE #39: Current log format is missing a useful date
**Is your feature request related to a problem? Please describe.**
The current log configuration includes the number of seconds from service start on every log line.  This timestamp is not useful because it cannot be easily determined when specific logs occurred.

**Describe the solution you'd like**
The easy solution is to change from `text` to `logfmt` handler type.

**Describe alternatives you've considered**
The code includes a comment to allow different formats, but supporting this requires a lot more work due to many formats having required configuration.

**What version of git2consul-go are you using?**

<pre>
$ git2consul -version
git2consul, version v0.1.2-dev (branch: master, revision: 9f404c06fe43479f3d81f237cee0ae93b3964905)
build date: 20200226-18:56:58
go version: go1.13.3
</pre>

**Additional context**
Switching to `logfmt` would result in more usable logs like:
```
$ ./git2consul                                                                                                                                                                            
timestamp=2020-02-26T10:57:33.644652-08:00 level=info message="Starting git2consul version: v0.1.2-dev"
timestamp=2020-02-26T10:57:33.64518-08:00 level=error message="No configuration file provided"
```

### PR DESCRIPTION
**Pull request by faizan82**
**Description**

Added option for structured log
Usage is displayed in case config file is not provided as an option

Fixes #[ISSUE-39](https://github.com/KohlsTechnology/git2consul-go/issues/39)

**Type of change**

* New feature (non-breaking change which adds functionality)

**Checklist**

- [ ] Unit tests updated: NA
- [+ ] Documentation updated

### CODE REVIEW

**Comment by Smiley73** [comment]
@faizan82 thanks for the PR. It looks good to me, I just want another set of eyes on it.
Apologies for moving so slow, unfortunately we're short staffed due to the pandemic.

File: main.go [hunk 1]
```diff
@@ -39,17 +40,30 @@ const (
...
 	flag.BoolVar(&printVersion, "version", false, "show version")
 	flag.BoolVar(&debug, "debug", false, "enable debugging mode")
 	flag.BoolVar(&once, "once", false, "run git2consul once and exit")
+	// allow switching logformat. Structured output helps with parsers
+	flag.StringVar(&logfmt, "logfmt", "text", "specify log format [text | json] ")
 	flag.Parse()
 
+	// Init checks
+	if len(filename) == 0 {
+		log.Errorf("No configuration file provided")
```
Comment (seanmalloy) [inline main.go:62]: I think this should use `log.Error` instead of `log.Errorf`.

File: main.go [hunk 2]
```diff
@@ -60,15 +74,16 @@ func main() {
...
-		log.Error("No configuration file provided")
-		os.Exit(ExitCodeFlagError)
+	switch logfmt {
+	case "text":
+		log.SetHandler(text.New(os.Stderr))
+	case "json":
+		log.SetHandler(json.New(os.Stderr))
 	}
 
+	//log.Infof("Starting git2consul version: %s", version.Version)
```
Comment (seanmalloy) [inline main.go:84]: Please remove this line of commented code.

Comment (faizan82) [reply main.go]: thank you @seanmalloy  removed the line

File: main.go (hunk 1 above)
Comment (faizan82) [reply main.go]: thank you @seanmalloy fixed this

File: main.go [hunk 3]
```diff
@@ -39,17 +40,30 @@ const (
...
+	// allow switching logformat. Structured output helps with parsers
+	flag.StringVar(&logfmt, "logfmt", "text", "specify log format [text | json] ")
 	flag.Parse()
 
+	// Init checks
+	if len(filename) == 0 {
+		log.Error("No configuration file provided")
+		flag.Usage()
+		os.Exit(ExitCodeFlagError)
+	}
```
Comment (seanmalloy) [inline main.go:65]: Moving this check above the the "if printVersion" check on lines 71 to 74 causes the below bug.

```
/git2consul -version
2020/05/21 21:48:15 error No configuration file provided
Usage of ./git2consul:
  -config string
    	path to config file
  -debug
    	enable debugging mode
  -logfmt string
    	specify log format [text | json]  (default "text")
  -once
    	run git2consul once and exit
  -version
    	show version
```
```

Comment (faizan82) [reply main.go:65]: oops. Let me fix this and may be write a few test cases to cover these cases as well for future changes

Comment (faizan82) [reply main.go:70]: @seanmalloy thank you for spending time on this PR. I have fixed this issue and added two checks in e2e pkg. Looks like coverage is showing diff for auth.go but I have not made any changes there so may be it's a false positive

**Pull request merged**